pkg/scalers/aws_sigv4.go

```diff
@@ -0,0 +1,102 @@
+package scalers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+// SigV4Config configures signing requests with SigV4.
+type SigV4Config struct {
+	Enabled bool   `yaml:"enabled,omitempty"`
+	Region  string `yaml:"region,omitempty"`
+}
+
+// Custom round tripper to sign requests
+type roundTripper struct {
+	signer *v4.Signer
+	region string
+}
+
+var (
+	// ErrAwsAMPNoAwsRegion is returned when "awsRegion" is missing from the config.
+	ErrAwsAMPNoAwsRegion = errors.New("no awsRegion given")
+)
+
+func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+
+	// Sign request
+	rt.signer.Sign(req, nil, "aps", rt.region, time.Now())
+
+	// Create default transport
+	transport := &http.Transport{}
+
+	// Send signed request
+	return transport.RoundTrip(req)
+}
+
+// NewSigV4RoundTripper returns a new http.RoundTripper that will sign requests
+// using Amazon's Signature Verification V4 signing procedure. The request will
+// then be handed off to the next RoundTripper provided by next. If next is nil,
+// http.DefaultTransport will be used.
+//
+// Credentials for signing are retrieving used the default AWS credential chain.
+// If credentials could not be found, an error will be returned.
+func NewSigV4RoundTripper(config *ScalerConfig, next http.RoundTripper) (http.RoundTripper, error) {
+	metadata, err := parseAwsAMPMetadata(config)
+	if err != nil {
+		return nil, err
+	}
+	awsCfg, err := getAwsConfig(context.Background(), metadata.awsRegion, metadata.awsAuthorization)
+	if err != nil {
+		return nil, err
+	}
+
+	triggerMetadata := config.TriggerMetadata
+	if triggerMetadata == nil {
+		return nil, fmt.Errorf("trigger metadata cannot be nil")
+	}
+
+	awsRegion := triggerMetadata["awsRegion"]
+	if awsRegion == "" {
+		return nil, fmt.Errorf("awsRegion not configured in trigger metadata")
+	}
+
+	cs, err := awsCfg.Credentials.Retrieve(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	creds := credentials.NewStaticCredentials(cs.AccessKeyID, cs.SecretAccessKey, "")
+	signer := v4.NewSigner(creds)
```

Comment:
can't you just do something like:
```suggestion
	signer := v4.NewSigner(awsCfg.Credentials)
```
I'm afraid in case of podIdentity what will happen with `cs.AccessKeyID, cs.SecretAccessKey`

Reply:
this works with the pod identity e2e testing for aws_managed_prometheus

Reply:
We have to use https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/amp as it's part of skd-v2.

You can init there client with [`NewFromConfig`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/amp#NewFromConfig) and it uses `aws.Config`